From: ohering@suse.de
Subject: Disable access to lowmem in a Xen PV guest
References: bnc#964342
Patch-mainline: never, upstream has not decided yet

Upstream may prefer a different approach.
Until then restore xenlinux behaviour.

---
 arch/x86/mm/init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/mm/init.c
+++ b/arch/x86/mm/init.c
@@ -637,7 +637,7 @@ void __init init_mem_mapping(void)
 int devmem_is_allowed(unsigned long pagenr)
 {
 	if (pagenr < 256)
-		return 1;
+		return !xen_pv_domain();
 	if (iomem_is_exclusive(pagenr << PAGE_SHIFT))
 		return 0;
 	if (!page_is_ram(pagenr))
